src/renderers/shared/reconciler/instantiateReactComponent.js

```diff
@@ -83,6 +106,8 @@ function instantiateReactComponent(node, parentCompositeType) {
       // represenations. I.e. ART. Once those are updated to use the string
       // representation, we can drop this code path.
       instance = new element.type(element);
+    } else if (isStatelessComponentType(element.type)) {
```

Comment:
Are you sure React should support this kind of definition? What is benefit of this over stateless functions and classes?